deploy-service/common/src/main/java/com/pinterest/deployservice/bean/EnvironBean.java

```diff
@@ -188,6 +191,32 @@
     @JsonProperty("ensureTrustedBuild")
     private Boolean ensure_trusted_build;
 
+    public void validate() throws Exception {
+        // A bunch of these fields will always be alphanumeric (with _ and -)
+        if(!this.env_name.matches("^[A-Za-z0-9_\\-]*$")) {
```

Comment:
Hi Jeremy, Seems we might be adding lot of duplicate checks. Can we wrap it into @validate annotation and use instead for setters?